### PR TITLE
Increase default enterprise limit

### LINKED ISSUE #683: set default no. enterprises ^ 1?

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Closes  #683. Supersedes PR #1846. (See #1846 for more details)

### CODE REVIEW

**Comment by sauloperez** [comment]
I don't understand why do we need to supersede. Wasn't there a conversation going on?

**Comment by myriamboure** [comment]
@sauloperez I think the main conversation was just about Angular as Pierre didn't know about it but I guess as Matt could solve the issue this conversation is not so important... The real conversation happened on the issue mainly https://github.com/openfoodfoundation/openfoodnetwork/issues/683 were we agreed to set up the limit to 5 and display error modal if wants to add more. 
But you said in the previous PR that there were some specs that seem to rely on the limit to be 1. No one reacted to that, do you want to elaborate here?

**Comment by sauloperez** [comment]
> But you said in the previous PR that there were some specs that seem to rely on the limit to be 1. No one reacted to that, do you want to elaborate here? 

It's fine, @Matt-Yorkley already fixed them.

**Comment by Matt-Yorkley** [comment]
@sauloperez I was in a rush and just wanted to work a commit or two into the original PR. I did think about doing a pull request across the branches, but it hadn't been touched in a few months, so I thought this would be quicker.

**Comment by sauloperez** [comment]
I'm just asking because I feel like this back and forth with PRs we sometimes do make it harder to keep track of them. You never know which PR is the right one. With many open PRs this becomes exponential.

File: config/locales/en.yml [hunk 1]
```diff
@@ -1867,6 +1867,7 @@ Please follow the instructions there to make your enterprise visible on the Open
     resolve_errors: Please resolve the following errors
     more_items: "+ %{count} More"
     admin:
+      enterprise_limit_reached: "You have reached your enterprise limit. Write to %{contact_email} if you need to increase it."
```
Comment (sauloperez) [inline config/locales/en.yml]: Users haven't seen this concept of enterprise limit before. I'd reword this to "You have reached the limit of enterprises per account. ..." or something alike.

File: spec/features/admin/enterprise_user_spec.rb [hunk 2]
```diff
@@ -19,7 +19,7 @@
   describe "creating an enterprise user" do
     context "with a limitted number of owned enterprises" do
       scenario "setting the enterprise ownership limit" do
-        user.enterprise_limit.should == 1
+        user.enterprise_limit.should == 5
```
Comment (sauloperez) [inline spec/features/admin/enterprise_user_spec.rb]: please, take the chance to use the RSpec `expect` syntax.

**Comment by Matt-Yorkley** [comment]
@sauloperez requested changes made in 0a9988b

**Comment by daniellemoorhead** [comment]
@Matt-Yorkley @enricostano @sauloperez it would be great if you could stage this in the UK or Katuma so that @myriamboure could test it. 

All going well then @enricostano could merge it.

giphy

**Comment by sauloperez** [comment]
Loving the gif culture adoption :trollface:

**Comment by sauloperez** [comment]
Deploying to staging.katuma.org... :clock1:

**Comment by sauloperez** [comment]
@Matt-Yorkley you need to rebase onto master for this branch to work. There's a missing column:

```
   (2.1ms)  SELECT COUNT(*) FROM "enterprises" WHERE "enterprises"."visible" = 't' AND (sells != 'none') AND (confirmed_at IS NOT NULL AND sells != 'unspecified')
PG::Error: ERROR:  column "confirmed_at" does not exist
LINE 1: ...rises"."visible" = 't' AND (sells != 'none') AND (confirmed_...
                                                             ^
: SELECT COUNT(*) FROM "enterprises"  WHERE "enterprises"."visible" = 't' AND (sells != 'none') AND (confirmed_at IS NOT NULL AND sells != 'unspecified')
Completed 500 Internal Server Error in 25.5ms
```

@myriamboure you'll see the 500 error snail page until we don't fix this, sorry.

**Comment by Matt-Yorkley** [comment]
@sauloperez Rebased and pushed.

**Comment by myriamboure** [comment]
Staged on French staging https://staging.openfoodfrance.fr/ but trying to solve some email issues :-o 
I already have a question that we can start discussing while this is fixed, testing notes wip here: https://docs.google.com/document/d/1xhoV-TmA_BH-gvHn8nSDoRIxOmPfTLwVxxRlOyt0sq0/edit#

In case 2 I tested "Existing users have all min 5 as entreprise limit number"  but I'm not sure about that. What do we do about the existing users who have 1 as entreprise limit? I suggest we do nothing as I guess by now they would have asked us if they would need, AND the message is pretty clear when they try to create a second one. So it means this 5 rule will apply only to new people registering on the platform. Is that fine with you ? @sstead @kirstenalarsen

**Comment by enricostano** [comment]
> In case 2 I tested "Existing users have all min 5 as entreprise limit number" 

From what I see in the proposed code we're not touching existing users. So I expect that they will keep what they had before. If that's the expected behavior, can we please test that we're not breaking anything for existing users with 1 (or whatever) limit?

**Comment by sstead** [comment]
I think it's ok that we limit new users to 5, and existing users stay the way they are.

**Comment by myriamboure** [comment]
Ok, so then it's all good, testing notes here: https://docs.google.com/document/d/1xhoV-TmA_BH-gvHn8nSDoRIxOmPfTLwVxxRlOyt0sq0/edit?usp=sharing

Existing users stay with the existing limit they had before, whether it be one or more because we gave them more. Ready to go for me.

**Comment by Matt-Yorkley** [comment]
Maybe it would be best to add a migration to bump the other users up to 5 at the same time?

**Comment by myriamboure** [comment]
Ah nooo @Matt-Yorkley ! We said we don't want the exiting users to change! So no migration needed, please remove your last modification, I said it was ready to go is it was :-)

**Comment by myriamboure** [comment]
@Matt-Yorkley can you remove the migration thing you added? As explained above we don't want to change the enterprise limit for existing users. Just waiting for that to request merging, all was good in tests ;-) Thanks!

**Comment by Matt-Yorkley** [comment]
@myriamboure Migration removed. :+1:
